### ISSUE TITLE
Add an option to flatten the scene hierachy when searching scene inspector

### ISSUE THREAD

**Issue by Arnklit** [godotengine/godot-proposals#10043]
### Describe the project you are working on

A game in production

### Describe the problem or limitation you are having in your project

When searching a complicated scene tree, especially in the remote view it is difficult to get an overview of all the hits because it's always displayed in the tree and the tree is collapsed.

Searching for vbox here for example just shows the two parents that contain vboxes.
![image](https://github.com/godotengine/godot-proposals/assets/4955051/10e54dac-bef3-4c75-b960-f1698fbdd1a5)

You have to expand all the nodes to see the found nodes and it's difficult to parse due to it being at different levels in the hierachy.
![image](https://github.com/godotengine/godot-proposals/assets/4955051/c88a1489-150d-4553-b6f0-d73cf1fad66c)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add an option to the Scene inspector view to flatten the scene when searching.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

An option like this.
![image](https://github.com/godotengine/godot-proposals/assets/4955051/db201d58-6afb-4a3f-8d59-d8f987bdfb61)
which would result in a search like this:
![image](https://github.com/godotengine/godot-proposals/assets/4955051/87c38184-d17a-4d70-a8c3-22e34f2fe3cc)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

This would be used often

### Is there a reason why this should be core and not an add-on in the asset library?

This would help with general usability and make more sense to have built in.

**Comment by RedMser** [comment]
It should probably include some way to disambiguate nodes with same names, similar to how file browser does "recent" and "favorites".

Maybe in a tooltip or as dim text after the node's name?

Otherwise, it's hard to work with the results when your nodes don't have very distinct names, as is the case in your example screenshot.

**Comment by KoBeWi** [comment]
Related: #8359

**Comment by Arnklit** [comment]
Also related https://github.com/godotengine/godot-proposals/issues/10008

**Comment by akien-mga** [comment]
Yeah I'm with @RedMser on this, a flat hierarchy of nodes which all have the same name seems quite confusing.

@Arnklit Could you describe a bit more what you would then *do* with this hierarchy? Finding the nodes is one thing, but since you're not finding where they are, just that *they are*, what do you do with them afterwards?

I can think of shift-click selecting all of them to do MultiNodeEdit in the inspector, or maybe circling through nodes of a very specific type to inspect a property, when this node type is unique enough that you don't really care where it is in the scene tree (e.g. a "RedRobot" enemy node).

---

If we implement this, when the option is enabled, there should likely be a label or something in the SceneTree when filtered that clarifies why the behavior is like this. E.g. `"Flatten Hierarchy on Search" is enabled.`

I'm worried that some users might enable this setting inadvertently, and then be very confused that as soon as they start filtering nodes they lose their whole scene tree hierarchy, and can't make heads or tails of where the nodes are.

**Comment by Arnklit** [comment]
Sorry for the delay on this.

I think these are all really good points and I don't think it works very well as it currently is in the PR where it's still a tree view, but some nodes are hidden so you can get a skewed idea of the hierachy, I also think that makes it more unclear whether you are in this mode than if the tree was just completely flattened as suggested in the proposal. I don't think it's neccessary to show a warning that you are in this mode if the view get's flattened when searching as that would make it clear you are in a filtered view.

So I know it's easy to make an example where this doesn't seem useful because the nodes are all called similar things, but in a project where you've actually named your nodes and you have really big scenes it is very useful to be able to have the scene view act more like a pure search box and only show the hits without the tree structure.

Example Scene Tree:
![Image](https://github.com/user-attachments/assets/f15f0f9c-f214-4bc4-9412-480893255ffc)

Current Behaviour without the option enabled:
![Image](https://github.com/user-attachments/assets/28958819-2556-42f4-8953-45af172d2445)

Current Behaviour of the PR with the option enabled:
![Image](https://github.com/user-attachments/assets/279678fa-427d-4819-ab74-3276062a1f54)
Very confusing as you still get a tree view, but both the top level and a level between the nodes are missing.

I think I'd prefer to show the entire path after the node, it won't matter that it's long in most cases, but you can scroll over and view it if you need to.

Suggested Behaviour with the option enabled:
![Image](https://github.com/user-attachments/assets/907d1101-b01d-46d0-91fd-05088617cffe)

We could also add the full path to the hover popup.
![Image](https://github.com/user-attachments/assets/8bb3ed9c-b122-4f7d-ad93-c00389c776d9)

**Comment by YeldhamDev** [comment]
I disagree. Showing the hierarchy adds context, and if every node was flat, the real confusion would come from having multiple nodes with the same name without said context.

Showing the path would do very little to solve this, as children with long paths would make the start of the path text (which in most cases would be the only part visible) pointless. Having to check the full path in the tooltip would just add attrition compared to just showing the hierarchy itself.
